object/src/meta.rs

```diff
@@ -178,57 +109,99 @@ impl FileMetaTable {
                     // Implementation Version Name
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 8 + elem_len as u32;
+                    group_length_remaining -= 8 + elem_len;
                     builder.implementation_version_name(text.decode(&v)?)
                 }
                 Tag(0x0002, 0x0016) => {
                     // Source Application Entity Title
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 8 + elem_len as u32;
+                    group_length_remaining -= 8 + elem_len;
                     builder.source_application_entity_title(text.decode(&v)?)
                 }
                 Tag(0x0002, 0x0017) => {
                     // Sending Application Entity Title
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 8 + elem_len as u32;
+                    group_length_remaining -= 8 + elem_len;
                     builder.sending_application_entity_title(text.decode(&v)?)
                 }
                 Tag(0x0002, 0x0018) => {
                     // Receiving Application Entity Title
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 8 + elem_len as u32;
+                    group_length_remaining -= 8 + elem_len;
                     builder.receiving_application_entity_title(text.decode(&v)?)
                 }
                 Tag(0x0002, 0x0100) => {
                     // Private Information Creator UID
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 8 + elem_len as u32;
+                    group_length_remaining -= 8 + elem_len;
                     builder.private_information_creator_uid(text.decode(&v)?)
                 }
                 Tag(0x0002, 0x0102) => {
                     // Private Information
-                    let mut v = vec![0; elem_len];
+                    let mut v = vec![0; elem_len as usize];
                     file.read_exact(&mut v)?;
-                    group_length_remaining -= 12 + elem_len as u32;
+                    group_length_remaining -= 12 + elem_len;
                     builder.private_information(v)
                 }
                 Tag(0x0002, _) => {
                     // unknown tag, do nothing
+                    // could be an unsupported or non-standard attribute
```

Comment:
Small nit, they're reserved
`Values of all Tags (0002,xxxx) are reserved for use by this Standard and later versions of DICOM.`
which mentioned below the table on http://dicom.nema.org/medical/dicom/current/output/chtml/part10/chapter_7.html#table_7.1-1

Reply:
Well yes, but with this comment I am keeping two cases in mind:

- the standard is extended to admit more file meta attributes, but the library does not know them yet
- a pseudo-DICOM producer invents its own attribute within the same group

In either case, nothing happens, but in the future we might be interested in logging it differently from the case below this one.